qutebrowser/keyinput/modeparsers.py

```diff
@@ -169,11 +172,56 @@ def __init__(self, win_id, mode, parent=None):
         """
         super().__init__(win_id, parent)
         self._read_config(mode)
+        self._orig_sequence = keyutils.KeySequence()
         self._mode = mode
+        self._ignore_next_key = False
 
     def __repr__(self):
         return utils.get_repr(self, mode=self._mode)
 
+    def handle(self, e, *, dry_run=False):
```

Comment:
This seems to duplicate a lot of the logic in `BaseKeyParser.match`, so I'm wondering whether there's a better way. Maybe `KeySequence` should have an `original` attribute where the original is stored when using `with_mappings` instead? Not sure how much easier this would make the code without seeing it though - what do you think?

Reply:
Here's a patch that adds a `without_mappings` method to `KeySequence`, to see how the code looks would look like. It's kind of clunky and incomplete though (even has pylint errors), because `KeySequence` is used pretty much as an immutable object.
If you want, I can rewrite the `KeySequence` class to be more mutable, but that would probably belong in it's own PR.
https://gist.github.com/jakanaka-evan/4973c4a4e46e8e3bdae57b17dba0e26c